controller/src/main/java/org/jboss/as/controller/AttributeDefinition.java

```diff
@@ -501,10 +501,20 @@ public ModelNode validateOperation(final ModelNode operationObject) throws Opera
      * @throws OperationFailedException if the value is not valid
      */
     public final void validateAndSet(ModelNode operationObject, final ModelNode model) throws OperationFailedException {
+        PathAddress pathAddress = PathAddress.pathAddress(operationObject.get(ModelDescriptionConstants.OP_ADDR));
```

Comment:
@jmesnil 

This concerns me a bit, as I think it's possible, maybe likely, that callers will pass in a custom operation object. Perhaps guard this with an 'operationObject.has(ModelDescriptionConstants.OP_ADDR)'.

Not 'hasDefined' as both DomainRootDefinition and ServerRootResourceDefinition have writable attributes 'name' and 'organization'

Reply:
What I described above is definitely a problem, e.g. the standard write-attribute handling:

https://github.com/wildfly/wildfly-core/blob/main/controller/src/main/java/org/jboss/as/controller/AbstractWriteAttributeHandler.java#L93

ProcessNameWriteAttributeHandler does something similar. And then there is ReadChildrenResourceHandler which is using AttributeDefinition for *parameters* not for config attributes. That's a common use of that class.

Reply:
I see your point.

This enhancement should not be done for configuring the management model and not for operation parameters.
Guarding against the presence of a `OP_ADDR` will ensure that we only trigger that feature when we have an attribute "bound" to a resource at that address.

Reply:
@bstansberry I've updated the code to verify the presence of an `OP_ADDR`. I did not change the deprecation log but your comment made me realise that when we are validating a deprecated *parameter* (so without an `OP_ADDR` the message will not provide information on the location of the deprecated parameter).
I let it as it is as most of our deprecated AttributeDefinitions are most likely attributes rather than parameters